kbase-extension/static/kbase/js/widgets/function_output/kbaseSampleSet.js

```diff
@@ -0,0 +1,228 @@
+            }
+
+            self.client = new GenericClient(Config.url('service_wizard'), {token: self.authToken()});
+            self.ws = new Workspace(Config.url('workspace'),{'token':self.authToken()});
+
+            self.$elem.append($('<div>').attr('align', 'center').append($('<i class="fa fa-spinner fa-spin fa-2x">')));
+
+            // 1) get stats, and show the panel
+            var basicInfoCalls = [];
+            basicInfoCalls.push(
+                Promise.resolve(self.ws.get_objects2({objects: [{'ref': this.obj_ref}]}))
+                    .then(function(obj) {
+                        self.ss_obj_data = obj['data'][0]['data']
+                        self.ss_obj_info = obj['data'][0]['info']
+                        self.link_ref = self.ss_obj_info[6] + '/' + self.ss_obj_info[0] + '/' + self.ss_obj_info[4];
+                    }));
+
+            Promise.all(basicInfoCalls)
+                .then(function() {
+                    self.renderBasicTable();
+                })
+                .catch(function(err) {
+                    console.error('an error occurred! ' + err);
+                    self.$elem.empty();
+                    self.$elem.append('Error' + JSON.stringify(err));
+                });
+
+            return this;
+        },
+
+        renderBasicTable: function() {
+            var self = this;
+            var $container = this.$elem;
+            $container.empty();
+
+            var $tabPane = $('<div>');
+            $container.append($tabPane);
+
+
+            // Build the overview table
+            var $overviewTable = $('<table class="table table-striped table-bordered table-hover" style="margin-left: auto; margin-right: auto;"/>');
+
+            function get_table_row(key, value) {
+                return $('<tr>').append($('<td>').append(key)).append($('<td>').append(value));
+            }
+
+            $overviewTable.append(get_table_row('KBase Object Name',
+                '<a href="/#dataview/'+self.link_ref + '" target="_blank">' + self.ss_obj_info[1] +'</a>' ));
+                // leave out version for now, because that is not passed into data widgets
+                //'<a href="/#dataview/'+self.link_ref + '" target="_blank">' + self.ss_obj_info[1] + ' (v'+self.ss_obj_info[4]+')'+'</a>' ));
+            $overviewTable.append(get_table_row('Saved by', String(self.ss_obj_info[5])));
+            $overviewTable.append(get_table_row('Number of Samples', self.ss_obj_data['samples'].length ));
+            $overviewTable.append(get_table_row('Description', self.ss_obj_data['description']));
+
+            self.metadata_headers = [];
+            // get the metadata_keys
+            self.client.sync_call('SampleService.get_sample', [{
+                id: self.ss_obj_data['samples'][0]['id']
+            }]).then(function(sample){
+                var node_tree = sample[0]['node_tree'][0]
```

Comment:
Is it possible to have an empty sample object? Or an empty `'node_tree'`? If so, that'll crash.

Reply:
right now it shouldn't be possible, but I'll add in the checks for it